File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.java
class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
         Arrays.sort(arr);
        if (arr.length ==0)
            return false;
        
        int difference = arr[1]-arr[0];
        
        for (int i=0; i< arr.length-1; i++){
            if ((arr[i+1] - arr[i]) != difference)
                return false;
            }
        return true;
    }
}

File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        sorted_arr = sorted(arr)
        if len(sorted_arr) ==0:
            return False
        
        difference = sorted_arr[1]-sorted_arr[0]
        
        for num in range(len(sorted_arr)-1):
            if (sorted_arr[num+1] - sorted_arr[num]) != difference:
                return False
        return True
            